docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: "3"
+
+services:
+  postgres:
+    image: postgres:11-alpine
+    environment:
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postmangovsg_dev
+    ports:
+      - "5432:5432"
+    volumes:
+      - pg-data:/var/lib/postgresql/data
+  redis:
+    image: redis:6-alpine
+    ports:
+      - "6379:6379"
+  localstack:
+    image: localstack/localstack:latest
+    ports:
+      - "4566-4599:4566-4599"
+    environment:
+      - DEBUG=1
+      - DATA_DIR=/tmp/localstack/data
+      - FILE_STORAGE_BUCKET_NAME=${FILE_STORAGE_BUCKET_NAME:-localstack-upload}
+      - AWS_LOG_GROUP_NAME=${AWS_LOG_GROUP_NAME:-postmangovsg-beanstalk-localstack}
+    volumes:
+      - ./localstack:/docker-entrypoint-initaws.d
+      - localstack-data:/tmp/localstack
+      - /var/run/docker.sock:/var/run/docker.sock
```

Comment:
Why do you need to mount the docker socket? This is quite risky as the containerized service now can make docker commands as if they were the docker host.

Reply:
@lamkeewei

Reply:
This is required if we would like to use the lambda functionality in localstack since it uses the host Docker daemon to execute lambda code in a container. 

Will remove it. Accidentally, left it there previously while experimenting with the localstack lambda stuff.

Reply:
Will it help us to debug lambdas?

Reply:
Personally, I feel that it's pretty tedious. For reference the steps required to run a lambda within localstack is as follow: 
1. Create lambda functions within localstack on startup (can be added into init scripts):
```
aws --endpoint-url http://localhost:4566 lambda create-function \
    --function-name unsusbcribe-digest \
    --code S3Bucket="__local__",S3Key="/local/path/to/serverless/unsubscribe-digest/build" \
    --handler index.handler \
    --runtime nodejs12.x \
    --role serverless
```
3. Make sure that Typescript code is compiled locally
2. To test invoke lambda function manually using AWS cli
```
aws --endpoint-url http://localhost:4566 \
    --function-name unsubscribe-digest \
    --payload @file://payload.json \
    response.json
```
3. Inspect `response.json` for the output

I feel that the only benefit for using localstack is the ability to simulate triggering of lambdas based on another action on AWS. For e.g., triggering a lambda when a file is uploaded on an S3 bucket. 

But for our case, there's currently no support to automatically trigger a lambda to simulate a bounce/complaint events when an email is sent. I think in the pro version there's a built in SMTP server that can trigger bounce/complaint events when a email is routed through it, but cannot confirm this. 

I was also looking at potentially running a local version of our API Gateway setup through localstack. But again, it gets pretty tedious to deploy onto localstack if you're not using something like [SAM](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html). You'll either have to write a super convoluted bash script to call the AWS cli commands to setup the resources/methods ([example](https://gist.github.com/crypticmind/c75db15fd774fe8f53282c3ccbe3d7ad)) or roll your own Cloudformation template to set it up. Both of which feels like overkill and would require developers to have knowledge of how to set this whole thing up.

Reply:
Agreed.  SAM requires you to write a convoluted template, and it's not easy to use. We would rather have a server with endpoints that our server can hit. 

> you can't just refer to the database running locally through localhost
Don't you refer to the database by using the service name defined in the docker-compose.yml file?

Reply:
>  Don't you refer to the database by using the service name defined in the docker-compose.yml file?

Yeah. But that would put a hard requirement to run all your services through Docker.